core/api/storage_miner/storage_api.cpp

```diff
@@ -86,11 +86,55 @@ namespace fc::api {
       return result;
     };
 
-    api->SectorsStatus = [=](SectorNumber id,
-                             bool) -> outcome::result<ApiSectorInfo> {
-      // TODO(ortyomka): [FIL-421] implement it
+    api->SectorsStatus =
+        [=](SectorNumber id,
+            bool show_onchain_info) -> outcome::result<ApiSectorInfo> {
       OUTCOME_TRY(sector_info, miner->getSealing()->getSectorInfo(id));
-      return ApiSectorInfo{.state = sector_info->state};
+
+      const auto &pieces{sector_info->pieces};
+      std::vector<DealId> deals;
+      deals.reserve(sector_info->pieces.size());
+
+      for (const auto &piece : pieces) {
+        deals.push_back(piece.deal_info ? piece.deal_info->deal_id : 0);      }
+      ApiSectorInfo api_sector_info{
+          sector_info->state,
+          id,
+          sector_info->sector_type,
+          sector_info->comm_d,
+          sector_info->comm_r,
+          sector_info->proof,
+          std::move(deals),
+          std::move(pieces),
```

Comment:
```suggestion
          pieces,
```

Reply:
it's const ref above. may remove const, or don't move

Reply:
There is shared_ptr, it can move sealing data. So `const` is ok, move is redundant.